zerver/webhooks/taiga/view.py

```diff
@@ -242,61 +261,64 @@ def parse_change_event(change_type: str, message: Mapping[str, Any]) -> Optional
-            "end_type": "test"
+        'type': 'webhook_test',
+        'event': 'test',
+        'values': {
+            'user': get_owner_name(message),
+            'user_link': get_owner_link(message),
+            'end_type': 'test'
         }
     }
 
 
 def parse_message(message: Mapping[str, Any]) -> List[Dict[str, Any]]:
-    """ Parses the payload by delegating to specialized functions. """
+    ''' Parses the payload by delegating to specialized functions. '''
     events = []
-    if message["action"] in ['create', 'delete']:
+    if message['action'] in ['create', 'delete']:
         events.append(parse_create_or_delete(message))
-    elif message["action"] == 'change':
-        if message["change"]["diff"]:
-            for value in message["change"]["diff"]:
+    elif message['action'] == 'change':
+        if message['change']['diff']:
+            for value in message['change']['diff']:
                 parsed_event = parse_change_event(value, message)
                 if parsed_event:
                     events.append(parsed_event)
-        if message["change"]["comment"]:
+        if message['change']['comment']:
             events.append(parse_comment(message))
-    elif message["action"] == "test":
+    elif message['action'] == 'test':
         events.append(parse_webhook_test(message))
 
     return events
 
 def generate_content(data: Mapping[str, Any]) -> str:
-    """ Gets the template string and formats it with parsed data. """
+    ''' Gets the template string and formats it with parsed data. '''
     template = templates[data['type']][data['event']]
     content = template.format(**data['values'])
-    end_type = 'end_type'
-    if template.endswith('{subject}**.'):
-        end_type = 'subject'
-    elif template.endswith('{epic_subject}**.'):
-        end_type = 'epic_subject'
-    elif template.endswith('{new}.') or template.endswith('{new}**.'):
-        end_type = 'new'
-    elif template.endswith('{old}.') or template.endswith('{old}**.'):
-        end_type = 'old'
-    end = data['values'].get(end_type)
-
-    if end[-1] in string.punctuation:
-        content = content[:-1]
-
     return content
 
 def get_owner_name(message: Mapping[str, Any]) -> str:
-    return message["by"]["full_name"]
+    return message['by']['full_name']
+
+def get_owner_link(message: Mapping[str, Any]) -> str:
+    return message['by']['permalink']
 
 def get_subject(message: Mapping[str, Any]) -> str:
-    data = message["data"]
-    return data.get("subject", data.get("name"))
+    data = message['data']
+    if 'permalink' in data:
+        return '['+data.get('subject', data.get('name'))+']'+'('+data['permalink']+')'
+    return '**'+data.get('subject', data.get('name'))+'**'
+
+def get_epic_subject(message: Mapping[str, Any]) -> str:
+    if 'permalink' in message['data']['epic']:
+        return'['+message['data']['epic']['subject']+']'+'('+message['data']['epic']['permalink']+')'
```

Comment:
Also, please include a space after `return`; it's certainly bad style (I'm surprised it even works without the space).